icicle/backend/cpu/src/curve/cpu_msm.hpp

```diff
@@ -599,24 +702,16 @@ eIcicleError cpu_msm(
   const Device& device, const scalar_t* scalars, const A* bases, int msm_size, const MSMConfig& config, P* results)
 {
   int c = config.c;
-  if (c < 1) { c = Msm<A, P>::get_optimal_c(msm_size, config.precompute_factor); }
-  if (scalar_t::NBITS % c == 0) {
-    ICICLE_LOG_WARNING << "MSM c (" << c << ") mustn't divide scalar width (" << scalar_t::NBITS
-                       << ") without remainder. Incrementing to a higher c";
-  }
-  while (scalar_t::NBITS % c == 0) {
-    c++;
-  }
+  if (c < 1) { c = std::max((int)std::log2(msm_size) - 1, 8); }
```

Comment:
either you are not rebased or you removed the get_optimal_c() call which is better

Reply:
where is it in Icicle?

Reply:
changed - also updated get_optimal_c to consider the value of precompute_factor.